tracardi/process_engine/action/v1/sort_action.py

```diff
@@ -0,0 +1,75 @@
+from pydantic import validator
+from tracardi.service.plugin.domain.config import PluginConfig
+from tracardi.domain.event import Event
+from tracardi.domain.session import Session
+from tracardi.domain.profile import Profile
+from tracardi.service.plugin.domain.register import Plugin, Spec, MetaData, Documentation, PortDoc, Form, FormGroup, \
+    FormField, FormComponent
+from tracardi.service.plugin.domain.result import Result
+from tracardi.service.plugin.runner import ActionRunner
+from tracardi.process_engine.tql.equation import MathEquation
+
+
+class SortConfig(PluginConfig):
+    data: str
+    direction: str
+
+    @validator("data")
+    def must_not_be_empty(cls, value):
+        if len(value) == 0:
+            raise ValueError("data is empty")
+        return value
+
+    @validator("direction")
+    def must_not_be_empty(cls, value):
+        if value not in ("asc", "desc"):
+            raise ValueError("direction is not \"asc\" or \"desc\"")
+        return value
+
+
+def validate(config: dict):
+    return SortConfig(**config)
+
+
+class SortAction(ActionRunner):
+    config: SortConfig
+
+    async def set_up(self, init):
+        self.config = validate(init)
+
+    async def run(self, payload: dict, in_edge=None) -> Result:
+        reverse = True if self.config.direction == "desc" else False
+        array = sorted(payload[self.config.data], reverse=reverse)
```

Comment:
This will work only for data in payload. It is better to use our so notation. 

'''
dot = self._get_dot_accessor(payload)
data = dot[self.config.data]
'''

And now sort data. 

More on to data references can be found here. http://docs.tracardi.com/plugins/tutorial/part3/ and here http://docs.tracardi.com/notations/dot_notation/